src/main/java/org/broadinstitute/barclay/argparser/TaggedArgumentParser.java

```diff
@@ -0,0 +1,333 @@
+            }
+        }
+    }
+
+    /**
+     * Attempt to retrieve a
+     * @param putativeSurrogateKey putative key to try to retrieve from the surrogate map
+     * @return tagged option pair for this surrogate, or null if no entry
+     */
+    public Pair<String, String> getTaggedOptionForSurrogate(final String putativeSurrogateKey) {
+        return tagSurrogates.get(putativeSurrogateKey);
+    }
+
+    // See if the opt parser would think this is a short option ("-")
+    private static boolean isShortOptionToken(final String argument) {
+        return argument.startsWith( SHORT_OPTION_PREFIX )
+                && !SHORT_OPTION_PREFIX.equals( argument )
+                && !isLongOptionToken( argument );
+    }
+
+    // See if the opt parser would think this is a long option ("--")
+    private static boolean isLongOptionToken(final String argument) {
+        return argument.startsWith( LONG_OPTION_PREFIX );
+    }
+
+    private String getSurrogateKeyForTaggedOption(
+            final String rawOptionString,   // the raw option string provided by the user, including the prefix and option name
+            final String rawArgumentValue,  // the raw argument value provided by the user
+            final String tagString          // the tag string that has been peeled off of the raw option string, including logical name and any attributes
+    )
+    {
+        final String surrogateKey = makeSurrogateKey(rawOptionString, rawArgumentValue);
+        if (null != tagSurrogates.put(surrogateKey, Pair.of(tagString, rawArgumentValue))) {
+            throw new CommandLineException.BadArgumentValue(
+                    String.format("The argument value: \"%s %s\" was duplicated on the command line", rawOptionString, rawArgumentValue));
+        }
+        return surrogateKey;
+    }
+
+    /**
+     * Construct a surrogate key from a raw option string (as provided by the user) and raw argument value (as
+     * provided by the user). In order to ensure uniqueness of the key, it needs to include all elements from the
+     * command line, including:
+     *
+     *  -the option prefix (short/long) used
+     *  -the option name (short/long) used
+     *  -the tag string used
+     *  -the argument value used
+     *
+     * @param rawOptionString the entire raw option string as provided by the user
+     * @param rawArgumentValue the entire raw argument value provided by the user
+     * @return a surrogate key that can be included as a substitute value in the command line args presented to
+     * command line the parser
+     */
+    private String makeSurrogateKey(
+            final String rawOptionString,
+            final String rawArgumentValue
+    ) {
+        // The keys strings are never parsed, but the separator is added to the
+        // middle to make visual key inspection easier
+        return rawOptionString + ARGUMENT_TAG_NAME_SEPARATOR + rawArgumentValue;
```

Comment:
Yes, confirmed that you are fine, since you rely on the `Pair` stored in the map for separation of the tag from the value, and don't parse the key. Disregard this comment chain.